src/_pytest/scope.py

```diff
@@ -0,0 +1,79 @@
+"""
+Scope definition and related utilities.
+
+Those are defined here, instead of in the 'fixtures', module because
+their use is spread across many other pytest modules, and centralizing it in 'fixtures'
+would cause circular references.
+
+Also this makes the module light to import, as it should.
+"""
+from enum import Enum
+from functools import lru_cache
+from typing import Optional
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing_extensions import Literal
+
+    _ScopeName = Literal["session", "package", "module", "class", "function"]
+
+
+class Scope(Enum):
+    """Represents one of the possible fixture scopes in pytest."""
+
+    Session = "session"
+    Package = "package"
+    Module = "module"
+    Class = "class"
+    Function = "function"
+
+    def index(self) -> int:
+        """Return this scope index. Smaller numbers indicate higher scopes (Session = 0)."""
+        return _scope_to_index(self)
+
+    def next(self) -> "Scope":
+        """Return the next scope (from top to bottom)."""
```

Comment:
bascially this one is the public api can can raise a better value error

Reply:
You were right in the other comment, this is used internally only. 👍